client/trino-client/src/main/java/io/trino/client/StatementClientV1.java

```diff
@@ -242,6 +181,63 @@ public boolean isFinished()
         return state.get() == State.FINISHED;
     }
 
+    @Override
+    public boolean validateCredentials(int timeout)
+            throws IOException
+    {
+        if (timeout <= 0) {
+            throw new IllegalArgumentException(String.format("Timeout is less than or equal to 0, %s <= 0", timeout));
+        }
+
+        Request request = buildValidationRequest();
+        Exception cause = null;
+        Duration timeoutNano = new Duration(timeout, TimeUnit.SECONDS);
+        long start = System.nanoTime();
+        long attempts = 0;
+
+        while (true) {
+            if (isClientAborted()) {
+                return false;
+            }
+
+            if (attempts > 0) {
+                Duration sinceStart = Duration.nanosSince(start);
+                if (sinceStart.compareTo(timeoutNano) > 0) {
+                    throw new IOException(format("ValidateCredentials timed out after %s. Last error: %s.", sinceStart,
+                            cause == null ? "null" : cause.getClass().getName()), cause);
+                }
+                try {
+                    MILLISECONDS.sleep(attempts * 100);
+                }
+                catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    return false;
+                }
+            }
+            attempts++;
+
+            try (Response response = httpCallFactory.newCall(request).execute()) {
+                switch (response.code()) {
+                    case HTTP_OK:
+                        return true;
+                    case HTTP_UNAUTHORIZED:
+                        return false;
+                    case HTTP_BAD_METHOD:
+                        throw new IOException("Target server does not support HEAD ./v1/statement");
```

Comment:
Need to distinguish 3 cases here: authentication success, authentication failure and couldn't authenticate (Exception). These will be used in `TrinoConnection<init>` and `TrinoConnection.isValid` will fold the 2nd and the 3rd cased into one and return `false`.

Reply:
When `TrinoConnection.isValid()` is called against an older version of Trino that does not support the `HEAD` call, we don't want that to invalidate the connection.  In that case we want the driver to behave like older versions of the driver, which don't make the `HEAD` call.

Reply:
I changed it to raise `UnsupportedOperationException` so `TrinoConnection.isValid` can be backward compatible.

For `TinoConnection` constructor,  it will error out if remote Trino server does not support this operation while client explicitly ask `validateConnection = true`.